pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -237,10 +238,26 @@ func (c *Cluster) failMon(monCount, desiredMonCount int, name string) {
 			logger.Errorf("failed to remove mon %q. %v", name, err)
 		}
 	} else {
+		if c.spec.DisruptionManagement.ManagePodBudgets {
```

Comment:
Instead of checking for ManagePodBudgets here, I'd suggest passing the setting to the methods on the pdb handler so they can enable or disable the pdb. It will make the code a bit simpler for the caller here.

Reply:
checking whether `ManagePodBudgets` is set inside the `allowMonDrain` and `blockMonDrain` functions.